climada/engine/impact.py

```diff
@@ -862,6 +865,77 @@ def write_col(i_col, imp_ws, xls_data):
 
         imp_wb.close()
 
+    def write_hdf5(self, file_data, todense=False):
+        """Write centroids attributes into hdf5 format.
+
+        Parameters
+        ----------
+        file_data : str or h5
+            If string, path to write data. If h5 object, the datasets will be generated there.
+        todense: bool
+            if True write the sparse matrices as hdf5.dataset by converting them to dense format
+            first. This increases readability of the file for other programs. default: False
+        """
+        if isinstance(file_data, (str, Path)):
+            LOGGER.info('Writing %s', file_data)
+            with h5py.File(file_data, 'w') as data:
+                self._write_hdf5(data, todense)
+        else:
+            self._write_hdf5(file_data, todense)
```

Comment:
I feel like the split in `write_hdf5` and `_write_hdf5` is cumbersome and supporting an already opened file is not necessary. I suggest renaming `file_data` to `file_path` and just allowing paths/strings here. Then both functions can be merged into one

Reply:
I agree and thanks for adapting accordingly.